File: src/main/java/de/noname/pizza4c/datamodel/lieferando/Product.java
package de.noname.pizza4c.datamodel.lieferando;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Product {
    @JsonProperty
    private String name;

    @JsonProperty
    private List<String> description;

    @JsonProperty
    private String imageUrl;

    @JsonProperty
    private List<Variant> variants;

    @JsonProperty
    private ProductInfo productInfo = new ProductInfo();

    public List<String> getDescription() {
        return description == null ? null : description.stream().map(s -> s.replace("• ", "")).toList();
    }


}

File: frontend/pizza4c-frontend/src/datamodel/restaurant/product.ts
import Variant from "./variant.ts";
import ProductInfo from "./productInfo.ts";
import {selectableVegan} from "../cart/vegan.ts";

export default interface Product {
    name: string;
    description: string[];
    imageUrl: string;
    variants: Variant[];
    productInfo: ProductInfo;
    isVegetarian: boolean;
    isVegan: boolean;
}

export function isVeganStateProduct(product: Product, veganState: selectableVegan): boolean {
    switch (veganState){
        case "all":
            return true;
        case "vegetarian":
            return product.isVegetarian;
        case "vegan":
            return product.isVegan;
    }
}